src/nomad_simulations/schema_packages/model_method.py

```diff
@@ -81,308 +87,574 @@ class ModelMethod(BaseModelMethod):
+    )
+    kind = Quantity(
+        type=MEnum('exchange', 'correlation', 'xc', 'k'), description='Component kind.'
+    )
 
-        | VdW method  | Reference                               |
-        | --------------------- | ----------------------------------------- |
-        | `"TS"`  | http://dx.doi.org/10.1103/PhysRevLett.102.073005 |
-        | `"OBS"` | http://dx.doi.org/10.1103/PhysRevB.73.205101 |
-        | `"G06"` | http://dx.doi.org/10.1002/jcc.20495 |
-        | `"JCHS"` | http://dx.doi.org/10.1002/jcc.20570 |
-        | `"MDB"` | http://dx.doi.org/10.1103/PhysRevLett.108.236402 and http://dx.doi.org/10.1063/1.4865104 |
-        | `"XC"` | The method to calculate the VdW energy uses a non-local functional |
-        """,
+    # Hybrid / range-separated parameters — PARSERS fill these if known
+    fraction_exact_exchange = Quantity(
+        type=np.float64, description='Global HF mixing α.'
+    )
+    sr_fraction_exact_exchange = Quantity(
+        type=np.float64, description='Short-range HF α_SR.'
+    )
+    lr_fraction_exact_exchange = Quantity(
+        type=np.float64, description='Long-range HF α_LR.'
+    )
+    range_separation_parameter = Quantity(
+        type=np.float64, unit='1/bohr', description='Range separation ω.'
     )
 
-    def __init__(self, m_def: 'Section' = None, m_context: 'Context' = None, **kwargs):
-        super().__init__(m_def, m_context, **kwargs)
-        self._jacobs_ladder_map = {
-            'lda': 'LDA',
-            'gga': 'GGA',
-            'mgg': 'meta-GGA',
-            'hyb_mgg': 'hyper-GGA',
-            'hyb': 'hybrid',
-        }
+    weight = Quantity(
+        type=np.float64, default=1.0, description='Weight in linear combination.'
+    )
 
-    def resolve_libxc_names(
-        self, xc_functionals: list[XCFunctional]
-    ) -> Optional[list[str]]:
-        """
-        Resolves the `libxc_names` and sorts them from the list of `XCFunctional` sections.
 
-        Args:
-            xc_functionals (list[XCFunctional]): The list of `XCFunctional` sections.
+class XCFunctional(ArchiveSection):
+    """
+    Normalized XC information for a calculation (possibly multi-component).
+    """
 
-        Returns:
-            (Optional[list[str]]): The resolved and sorted `libxc_names`.
-        """
-        return sorted(
-            [
-                functional.libxc_name
-                for functional in xc_functionals
-                if functional.libxc_name is not None
-            ]
-        )
+    components = SubSection(sub_section=XCComponent.m_def, repeats=True)
 
-    def resolve_jacobs_ladder(
-        self,
-        libxc_names: list[str],
-    ) -> str:
-        """
-        Resolves the `jacobs_ladder` from the `libxc_names`. The mapping (libxc -> NOMAD) is set in `self._jacobs_ladder_map`.
+    raw_string = Quantity(
+        type=str,
+        description="Original XC string from the code (e.g. 'HSE06', 'SCAN+rVV10').",
+    )
```

Comment:
Very nice! I would just remove `XC`, as it sounds as if you're limiting yourself to libxc strings.

```suggestion
    raw_string = Quantity(
        type=str,
        description="Original functional name as reported in the code (e.g. 'HSE06', 'SCAN+rVV10').",
    )
```

Reply:
You should also define the value in case it's a user-defined functional. Do we leave it `None` then?

Reply:
okay, this quantity was removed instead